wagtail/snippets/tests/test_viewset.py

```diff
@@ -769,6 +770,47 @@ def test_boolean_column(self):
         )
 
 
+class TestRelatedFieldColumns(BaseSnippetViewSetTests):
+    model = SnippetChooserModel
+
+    def setUp(self):
+        super().setUp()
+        url = "https://example.com/free_examples"
+        self.advert = Advert.objects.create(url=url, text="Free Examples")
+        self.ffs = FullFeaturedSnippet.objects.create(
+            text="test snippet with custom icon"
+        )
+        self.scm = self.model.objects.create(advert=self.advert, full_featured=self.ffs)
+        self.no_ffs_chooser = self.model.objects.create(advert=self.advert)
```

Comment:
It's best to move the creation of this object to the test for empty FK specifically, as otherwise the other unrelated tests could also fail if the implementation didn't handle the empty FK case.